tests/integration_tests/ui_tests/ArelleGUITest/ArelleGUITest/Tests.cs

```diff
@@ -0,0 +1,373 @@
+                // +1 for the header, +0.5 to end up in the middle
+                + selectFileFromArchivePane.BoundingRectangle.Height * (indexOfEntryPoint + 1.5) / (nFilesInDocumentZip + 1))
+            ));
+            Keyboard.Type(VirtualKeyShort.TAB, VirtualKeyShort.ENTER);
+
+            Logger.Default.Info("wait for document to load and UI to populate");
+            Retry.WhileNull(() => Window.FindFirstByXPath("/Pane/Pane/Pane/Pane/Pane/Pane/Pane"), timeout: TimeSpan.FromMinutes(1), ignoreException: true);
+            var iconsAndMainArea = Window.FindFirstByXPath("/Pane/Pane");
+            var iconsAndMainAreaChildren = iconsAndMainArea.FindAllChildren();
+            var mainArea = iconsAndMainAreaChildren[0];
+            var icons = iconsAndMainAreaChildren[1];
+            var statusBar = iconsAndMainAreaChildren[2];
+            var mainAreaChildren = mainArea.FindAllChildren();
+            var messagesAreaAndTabBar = mainAreaChildren[0];
+            var tablesAndFactView = mainAreaChildren[1];
+            var messagesArea = messagesAreaAndTabBar.FindFirstChild();
+            var tablesAndFactViewChildren = tablesAndFactView.FindAllChildren();
+            var factViewAndTabBar = tablesAndFactViewChildren[0];
+            var tablesAndTabBar = tablesAndFactViewChildren[1];
+            var factView = factViewAndTabBar.FindFirstChild();
+            var tables = tablesAndTabBar.FindFirstChild();
+
+            Logger.Default.Info("copy structure to clipboard");
+            Mouse.RightClick(tables.BoundingRectangle.Center());
+            Wait.UntilInputIsProcessed();
+            var copyToClipboardMenuItem = Window.ContextMenu.Items[4];
+            copyToClipboardMenuItem.Click();
+            Wait.UntilInputIsProcessed();
+            var copyToClipboardSubMenu = Automation.GetDesktop()
+                .FindAllChildren(cf => cf.ByControlType(ControlType.Menu).And(cf.ByProcessId(Application.ProcessId)))
+                .Select(e => e.AsMenu())
+                .Single(e => e.Items.Length == 3);
+            var copyTableMenuItem = copyToClipboardSubMenu.Items[2];
+            copyTableMenuItem.Click();
+            Wait.UntilInputIsProcessed();
+            Thread.Sleep(200);
+            Logger.Default.Info("get clipboard text");
+            var tableText = GetClipboardText();
+            Assert.That(tableText, Is.EqualTo(TABLE_STRUCTURE.Replace("\r", "")));
+        }
+
+        static int? GetParentProcessId(Process process)
+        {
+            return (int?)typeof(Process)
+                    .GetProperty("ParentProcessId", System.Reflection.BindingFlags.NonPublic | System.Reflection.BindingFlags.Instance)!
+                    .GetGetMethod(true)!
+                    .Invoke(process, null);
+        }
+
+        static string GetClipboardText()
+        {
+            string s = "";
+            var t = new Thread(() => { s = System.Windows.Forms.Clipboard.GetText(); });

```

Comment:
Why does this need to be multithreaded? It seems like it immediately waits anyways?

Reply:
I am not sure what that means, but [this](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.clipboard.gettext?view=windowsdesktop-7.0) doesn't imply to me that you should need a thread if your main thread is setting the correct sta attribute.